Actions toggleable on List Create Controller.

zahnno opened the pull request:
This pr provides the ability to pass in a boolean property (showActions) to the list create controller, which then hides actions associated to list create controller.

Commit:
Copied over existing PostgresDAO implementation

Commit:
Modified prepareStatement to return void and take in a prepared statement

Commit:
Made ConnectionPool not abstract. Removed static modifiers from functions. Modified static calls

Commit:
Removed SQLData. Updated remove_ function

Commit:
Updated remove_ function

Commit:
Updated find_ to use ThreadLocal StringBuilder

Commit:
Added fetching a list of columns to constructor

Commit:
Added buildFormattedColumnNames and buildFormattedColumnPlaceholders

Commit:
Updated compiler settings to use 1.8 instead of 1.7

Commit:
Made modifications to put_ to work like an upsert command.

Commit:
Fixed typo

Commit:
Added throwing of UnsupportedOperationException for removeAll_

Commit:
Added sqlType to primitive types

Commit:
Added getSqlType function to PropertyInfo interface. Added sql types to classes that implement it

Commit:
Added table creation upon PostgresDAO initialize. Removed sqlType refinements from refinements.js

Commit:
Changed id type to be serial

Commit:
Added closing of result sets, prepared statements, and connections

Commit:
Added SQLType enum

Commit:
Removed columns property and just using properties from FObject to avoid any issues with property ordering

Commit:
Filtered out id field from property info

Commit:
Removed where statement

Commit:
Changed serial to bigserial

Commit:
Modified createFObject function

Commit:
Merge branch 'master' of github.com:eatonk/foam2 into postgresDAO

Commit:
Merge branch 'master' of github.com:eatonk/foam2 into postgresDAO

zahnno committed:
show actions toggleable

zahnno committed:
removed save and back actions

Commit:
Modified PostgresDAO to make use of getStorageTransient functions

Commit:
Added VARBINARY sql type. Modified createFObject function

Commit:
Added logic to prevent from selecting data from out of bounds

Commit:
Added comments

Commit:
Added conditional for Byte array

Commit:
Added toLowerCase() when building column names

Commit:
Moved checking of storageTransient to above fetching of value

kgrgreer commented on src/foam/u2/ListCreateController.js:
```diff
@@ -155,7 +157,8 @@ foam.CLASS({
         function initE() {
-          this.tag(this.detailView, {data: this.obj, controllerMode: foam.u2.ControllerMode.VIEW}).add(this.BACK);
+          this.tag(this.detailView, {data: this.obj, controllerMode: foam.u2.ControllerMode.VIEW})
+          if (this.showActions) this.add(this.BACK);          
```
Conform to FOAM style guid with spaces around the parens.

kgrgreer commented on src/foam/u2/ListCreateController.js:
```diff
@@ -123,7 +124,8 @@ foam.CLASS({
           var view = this.createDetailView ? this.createDetailView : this.detailView
-          this.tag(view, {data: this.obj}).add(this.CANCEL, this.SAVE);
+          this.tag(view, {data: this.obj})
+          if (this.showActions) this.add(this.CANCEL, this.SAVE);
```
Conform to FOAM style guid with spaces around the parens.

kgrgreer commented:
Fix merge conflicts.

kgrgreer committed:
From the command line you can specify -Dfoam.main=myScript to run myScript on startup.

Commit:
Fixed id casting with find command. Fixed issue with not calling next on ResultSet

Commit:
Modified Long casting in remove_ function

Commit:
Simplified empty result set logic

Commit:
Modified handling of closing resources

Commit:
Fixed spacing

Commit:
Added TODO to handle non-number values

Commit:
Added a cache of non storage transient properties to avoiding creating a list everytime

Commit:
Modelled connection pool

Commit:
Fetching connectionPool from context

mdittmer committed:
Web Fetch API: Deal with missing headers API on Edge (#674)

adamvy committed:
BlobStoreCleanup: Return Identified blob, not the id.

Commit:
Catching Throwabes instead of Exceptions

adamvy committed:
Use ReturnBox for JS WebSockets as well.

adamvy committed:
Merge branch 'master' of github.com:foam-framework/foam2

Commit:
Added enabled = 't'

Commit:
Changed poolSize to 20 and added a closeAllQuietly function

Commit:
Added transient: true

Commit:
Removed SQLType

Commit:
Removed getSQLType function that made use of old modelled version of SQLType

kgrgreer committed:
Merge pull request #673 from eatonk/postgresDAO

PostgresDAO

Commit:
Added sqlType to refinements

Commit:
Added StringArray

Commit:
Added networkTransient and storageTransient

Commit:
Replaced setting of getSQLType java method with an expression for the sqlType property

kgrgreer committed:
Merge pull request #676 from eatonk/sqlTypes

SQL types

Commit:
Removed transient property in favor of just using either storageTransient or networkTransient properties instead

Commit:
Removed value

Commit:
Removed check of storage/network transient from toXML function

Commit:
Modelled OutputerMode. Changed output logic to check against OutputterMode instead of just the property

kgrgreer committed:
Merge pull request #677 from eatonk/removeTransient

Removed transient property

Commit:
Modelled PStream interface. Added apply method

Commit:
Modelled Parser. Modified Parser implementations to use PStream apply method

Commit:
Merge branch 'master' of github.com:foam-framework/foam2 into pstreamModifications

Commit:
Changed StringPS to StringPStream

Commit:
Changed references of StringPS to StringPStream

Commit:
Changed StringPS to StringPStream in JS code

Commit:
Removed package-lock.json

kgrgreer committed:
Merge pull request #678 from eatonk/pstreamModifications

PStream modifications

Commit:
Added apply function

Commit:
Modified .parse calls to use apply instead

Commit:
Modified more function calls to use apply instead of parse

Commit:
Replaced two more instances of parse() with apply()

kgrgreer committed:
Merge pull request #680 from eatonk/StringPStream_JS

StringPStream modifications

mdittmer committed:
Use foam.lookup() for core services in (potential) box contexts (#658)

A `foam.box.Context` that uses a `foam.box.ClassWhitelistContext` should not
be forced to whitelist core service classes (that do not need to be
(de)serialized and transported) in order to work. This change eliminates
`foam.net.node.Socket` and `foam.json.Outputter` instantiations that are
often invoked in a `foam.box.Context`. These instantiations now use
`foam.lookup()` instead.

mdittmer committed:
ForkBox.critical ==> exit when child exits (#659)

zahnno committed:
style guides
